pandas/tests/frame/test_convert_to.py

```diff
@@ -172,3 +172,9 @@ def test_to_records_index_name(self):
         df.index.names = ['A', None]
         rs = df.to_records()
         self.assertIn('level_0', rs.dtype.fields)
+
+    def test_to_records_with_unicode_index(self):
+        res = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a').to_records()
+        expect_res = np.rec.array([('x', 'y')], dtype=[('a', 'O'), ('b', 'O')])
+        self.assertEqual(res, expect_res)
```

Comment:
use 
```
result = ....
expected = ....
tm.assert_numpy_array_equal(result, expected)
```

use ``tm.assert_numpy_array_equal`` here (and only need 1 assert; that will compare everything)

add the issue number as a comment